crates/core/tedge_agent/src/lib.rs

```diff
@@ -60,6 +60,10 @@ pub struct AgentOpt {
     /// MQTT root prefix
     #[clap(long)]
     pub mqtt_topic_root: Option<Arc<str>>,
+
+    /// Interval at which the memory usage is logged (in seconds)
+    #[clap(long = "log-memory-interval", default_value = "60")]
+    pub log_memory_interval: u64,
```

Comment:
Is this a short term addition or something permanent? If permanent, any specific reason why this is a command-line argument and not tedge config setting (other than convenience and simplicity)? No objections here, but just trying to understand the rationale behind that choice.

Reply:
The idea is to set this right when the process starts - ideally before any thin-edge processing.

I see this as a new feature. This can possibly be extended with:
- the number of threads (currently we use the default i.e one thread per core),
- a cap on allocated memory (crashing if the process starts to go wild).

Reply:
Supporting a config would make it easier to enable on devices under test...as adding a flag is more difficult as we would need to use `systemctl edit ...` to enable this feature.

Reply:
> Supporting a config would make it easier to enable on devices under test...as adding a flag is more difficult as we would need to use `systemctl edit ...` to enable this feature.

Done with https://github.com/thin-edge/thin-edge.io/pull/2693#discussion_r1484348307

The new setting is `run.log_memory_interval` with a default of 5 minutes.